code/campaign/src/main/java/com/adobe/marketing/mobile/campaign/LocalNotificationHandler.java

```diff
@@ -0,0 +1,286 @@
+import androidx.core.content.ContextCompat;
+
+import com.adobe.marketing.mobile.MobileCore;
+import com.adobe.marketing.mobile.services.DeviceInforming;
+import com.adobe.marketing.mobile.services.Log;
+import com.adobe.marketing.mobile.services.ServiceProvider;
+import com.adobe.marketing.mobile.util.StringUtils;
+import java.security.SecureRandom;
+import java.util.HashMap;
+
+@SuppressWarnings("unchecked")
+public class LocalNotificationHandler extends BroadcastReceiver {
+
+    private static final String LOG_TAG = "Campaign";
+    private static final String SELF_TAG = "LocalNotificationHandler";
+    private static final String NOTIFICATION_CHANNEL_NAME = "ADOBE_EXPERIENCE_PLATFORM_SDK";
+    private static final String NOTIFICATION_CHANNEL_ID = "ADOBE_EXPERIENCE_PLATFORM_SDK";
+    private static final String NOTIFICATION_CHANNEL_DESCRIPTION =
+            "Adobe Experience Platform SDK Notifications";
+    private static final String NOTIFICATION_CONTENT_KEY = "NOTIFICATION_CONTENT";
+    private static final String NOTIFICATION_USER_INFO_KEY = "NOTIFICATION_USER_INFO";
+    private static final String NOTIFICATION_IDENTIFIER_KEY = "NOTIFICATION_IDENTIFIER";
+    private static final String NOTIFICATION_DEEPLINK_KEY = "NOTIFICATION_DEEPLINK";
+    private static final String NOTIFICATION_SOUND_KEY = "NOTIFICATION_SOUND";
+    private static final String NOTIFICATION_SENDER_CODE_KEY = "NOTIFICATION_SENDER_CODE";
+    private static final int NOTIFICATION_SENDER_CODE = 750183;
+    private static final String NOTIFICATION_REQUEST_CODE_KEY = "NOTIFICATION_REQUEST_CODE";
+    private static final String NOTIFICATION_TITLE = "NOTIFICATION_TITLE";
+
+    @Override
+    public void onReceive(final Context context, final Intent intent) {
+        // get message and request code from previous context
+        Bundle bundle = intent.getExtras();
+
+        if (bundle == null) {
+            Log.debug(LOG_TAG,
+                    SELF_TAG,
+                    "Failed to load extras from local notification intent");
+            return;
+        }
+
+        Context appContext = context.getApplicationContext();
+        String message;
+        String messageID;
+        String deeplink;
+        HashMap<String, Object> userInfo;
+        String sound;
+        int requestCode;
+        int senderCode;
+        String title;
+
+        try {
+            message = bundle.getString(NOTIFICATION_CONTENT_KEY);
+            requestCode = bundle.getInt(NOTIFICATION_REQUEST_CODE_KEY);
+            senderCode = bundle.getInt(NOTIFICATION_SENDER_CODE_KEY);
+            messageID = bundle.getString(NOTIFICATION_IDENTIFIER_KEY);
+            deeplink = bundle.getString(NOTIFICATION_DEEPLINK_KEY);
+            sound = bundle.getString(NOTIFICATION_SOUND_KEY);
+            userInfo = (HashMap<String, Object>) bundle.getSerializable(NOTIFICATION_USER_INFO_KEY);
+            title = bundle.getString(NOTIFICATION_TITLE);
+        } catch (Exception e) {
```

Comment:
does this exception get thrown if the bundle.getInt or getString returns a null value? if so, we should provide default values to prevent this exception from being thrown if not all local notification settings are provide in the in-app payload

Reply:
It is thrown if the type of the value is different from the one we are expecting (string, int etc.) It does return a default value, based on the type, if the key is not found
https://developer.android.com/reference/android/os/BaseBundle#getString(java.lang.String)